src/MainComponent.js

```diff
@@ -3,13 +3,20 @@ import CategoryList from "./CategoryList";
 import NoteList from "./NoteList";
 import Note from "./Note";
 import { useParams } from "react-router-dom";
+import {MainContext} from "./Utils";
 
 const MainComponent = () => {
     const {categoryId, noteId} = useParams();
     return(
         <Fragment>
-            <CategoryList categoryId={categoryId}/>
-            <NoteList noteId={noteId}/>
+            <MainContext.Provider value={categoryId}>
+                <CategoryList/>
+            </MainContext.Provider>
+
+            <MainContext.Provider value={noteId}>
+                <NoteList/>
+            </MainContext.Provider>
+
```

Comment:
Lets have object as context value, there is no need of having 2 providers of the same context
```suggestion
            <MainContext.Provider value={{categoryId, noteId}}>
                <CategoryList/>
                <NoteList/>
            </MainContext.Provider>

```